config: bump base GC time to 25 hours

madelynnblue opened the pull request:
The motivation for this change is to help new users correctly
support their own cluster. Say a new users starts up a new
cluster and begins putting data into it. They would like to
have backups, so they start taking daily (full) backups. They
quickly notice these take a long time, and would like to move
to something incremental. They read the docs about [incremental
backups](https://www.cockroachlabs.com/docs/stable/backup.html#incremental-backups)
which say:

> You can only create incremental backups within the garbage collection
period of the base backup's most recent timestamp.

The user clicks the link to the zone docs and quickly finds out
the default GC interval is 24 hours and makes the obvious (though
incorrect) assumption that they can safely perform incremental backups
every 24 hours. This assumption is incorrect because "within" from
the docs above means that the backup process, which is made up of many
queued requests, must start its last request before the GC interval.

The equation relating these things together is:

`GC interval = (desired backup interval) + (time to perform incremental
backup) + (fudge factor)`

So if users want to do a daily backup, they need the GC interval to
be 24h + something, and for new clusters an hour is almost certainly
good enough. We assume that as clusters become larger, the operators
will be more familiar with the system, will read more docs, and be
aware of this above equation (which we also need to document in the
incremental backup section, since it's not very obvious). But for most
people running a cluster, we'd like them to be successful by default.

We are aware that there are good arguments for reducing the default
GC time, and those can certainly be discussed elsewhere. However we
think that changing this time from 24h to 25h doesn't have any real
impact on any of those decisions since it is such a small change,
and so this PR shouldn't be blocked because of that uncertainty.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/17366)
<!-- Reviewable:end -->

tschottdorf commented:
LGTM, but add an abbreviated version of your rationale in a code comment.

madelynnblue committed:
config: bump base GC time to 25 hours

The motivation for this change is to help new users correctly
support their own cluster. Say a new users starts up a new
cluster and begins putting data into it. They would like to
have backups, so they start taking daily (full) backups. They
quickly notice these take a long time, and would like to move
to something incremental. They read the docs about [incremental
backups](https://www.cockroachlabs.com/docs/stable/backup.html#incremental-backups)
which say:

> You can only create incremental backups within the garbage collection
period of the base backup's most recent timestamp.

The user clicks the link to the zone docs and quickly finds out
the default GC interval is 24 hours and makes the obvious (though
incorrect) assumption that they can safely perform incremental backups
every 24 hours. This assumption is incorrect because "within" from
the docs above means that the backup process, which is made up of many
queued requests, must start its last request before the GC interval.

The equation relating these things together is:

`GC interval = (desired backup interval) + (time to perform incremental
backup) + (fudge factor)`

So if users want to do a daily backup, they need the GC interval to
be 24h + something, and for new clusters an hour is almost certainly
good enough. We assume that as clusters become larger, the operators
will be more familiar with the system, will read more docs, and be
aware of this above equation (which we also need to document in the
incremental backup section, since it's not very obvious). But for most
people running a cluster, we'd like them to be successful by default.

We are aware that there are good arguments for reducing the default
GC time, and those can certainly be discussed elsewhere. However we
think that changing this time from 24h to 25h doesn't have any real
impact on any of those decisions since it is such a small change,
and so this PR shouldn't be blocked because of that uncertainty.

mjibson commented:
Added a comment. Will merge later today.
